doc/lua_api.md

```diff
@@ -6109,8 +6132,8 @@ Utilities
   the given "time of day" value (as returned by `core.get_timeofday`).
 * `core.encode_png(width, height, data, [compression])`: Encode a PNG
   image and return it in string form.
-    * `width`: Width of the image (integer, range [0, 4294967295])
-    * `height`: Height of the image (integer, range [0, 4294967295])
+    * `width`: Width of the image (integer in range [u32])
```

Comment:
the code probably does not require this at the moment because it sidesteps any actual image classes, but just to be safe, i think we should also restrict this to imagesize.

because, well, this is an image, and it makes little sense to encode images with > 20k pixels in each direction this way...

Reply:
Rejected.

This is just encoding, not rendering and in my own tests this function manages to deal with very large pixel values just fine (but it will of course take more time). We should expect Luanti to be able to manage large amounts of data as well unless there's a bug.

`[imagesize]` is speficially for texture modifiers only and was added because Luanti itself added this restriction due to technical constraints (not performance).